client/my-sites/stats/geochart/index.jsx

```diff
@@ -283,8 +285,49 @@ class StatsGeochart extends Component {
 			return chartData;
 		}
 
-		// Default to country
-		chartData.addColumn( 'string', translate( 'Country' ).toString() );
+		// Region mode: groups regions by country, showing aggregated country data
+		// with tooltips displaying the top regions within each country.
+		// @TODO: once we have the region coordinates available through the API, we could
+		// get rid of this block and use the same as city mode.
+		if ( geoMode === 'region' ) {
+			chartData.addColumn( 'string', 'Location' );
+			chartData.addColumn( 'number', 'Views' );
+			chartData.addColumn( { type: 'string', role: 'tooltip', p: { html: true } } );
+
+			const groupedLocations = data.reduce( ( acc, location ) => {
```

Comment:
Although I usually default to `for` or `forEach` for similar cases, this is a nice usage of `reduce`. 🙂